tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -270,6 +272,30 @@ private static unsafe int ResolveManagedCallstack(int threadId, IntPtr context,
         }
     }
 
+    [UnmanagedCallersOnly]
+    private static unsafe int UnhandledExceptionFilter(IntPtr exceptionInfo)
+    {
+        // A lot of the logic is implemented on the native side (Datadog.Profiler.Native and libdatadog).
+        // Unfortunately, native exceptions such as access violation can't be caught on the managed side.
+        // It means that if the native side crashes, we don't have a chance to clean the ClrMD context,
+        // which in turn means that the threads of the crashing process will remain suspended, preventing it from exiting.
+        // To prevent that, we bypass .NET exception management and use SetUnhandledExceptionFilter to catch all exceptions.
+
+        // We're running managed code in a completely unsupported state, so keep the amount of work to an absolute minimum.
+        _dataTarget?.Dispose();
```

Comment:
Is there any way this could have already been disposed? And what happens if so?

Reply:
It's fine, there's a boolean check: https://github.com/microsoft/clrmd/blob/main/src/Microsoft.Diagnostics.Runtime/DataTarget.cs#L88-L103